src/main.rs

```diff
@@ -24,33 +23,68 @@ mod update;
 mod operation;
 mod screen;
 mod view;
+mod errors;
 
 use core::Core;
 use input::Input;
 use screen::Screen;
+use error_chain::ChainedError;
+
+use errors::*;
 
 fn main() {
+    if let Err(ref e) = run() {
+        use std::io::Write;
+        let stderr = &mut ::std::io::stderr();
+
+        writeln!(stderr, "error: {}", e).unwrap();
+
+        for e in e.iter().skip(1) {
+            writeln!(stderr, "caused by: {}", e).unwrap();
+        }
+
+        if let Some(backtrace) = e.backtrace() {
+            writeln!(stderr, "backtrace: {:?}", backtrace).unwrap();
+        }
+        ::std::process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
     log4rs::init_file("log_config.yaml", Default::default()).unwrap();
-    let xi = clap_app!(xi =>
-                       (about: "The Xi Editor")
-                       (@arg core: -c --core +takes_value "Specify binary to use for the backend")
-                       (@arg file: +required "File to edit")
-                      );
+    let xi = clap_app!(
+        xi =>
+        (about: "The Xi Editor")
+        (@arg core: -c --core +takes_value "Specify binary to use for the backend")
+        (@arg file: +required "File to edit"));
+
     let matches = xi.get_matches();
     let core_exe = matches.value_of("core").unwrap_or("xi-core");
     let file = matches.value_of("file").unwrap();
-    let mut core = Core::new(core_exe, file);
-    let mut screen = Screen::new();
+    let mut core = Core::new(core_exe);
+    let mut screen = Screen::new()?;
     let mut input = Input::new();
     input.run();
-    screen.init();
-    core.scroll(0, screen.size.1 as u64 - 2);
+    screen.init()?;
+    core.open(file)?;
+    core.scroll(0, screen.size.1 as u64 - 2)?;
 
     loop {
         if let Ok(event) = input.try_recv() {
-            input::handle(&event, &mut core);
+            if let Err(e) = input::handle(&event, &mut core) {
+                log_error(e);
+            }
         } else {
-            screen.update(&mut core);
+            if let Err(e) = screen.update(&mut core) {
+                log_error(e);
+            }
         }
     }
 }
+
+fn log_error<E: ChainedError>(e: E) {
```

Comment:
maybe at some point but for now it's used only in `main.rs`, so I think it makes sense to have it here.